src/commands/serve/serve.mjs

```diff
@@ -0,0 +1,182 @@
+
+  const { addonsUrls, capabilities, siteUrl, timeouts } = await getSiteInformation({
+    // inherited from base command --offline
+    offline: options.offline,
+    api,
+    site,
+    siteInfo,
+  })
+
+  /** @type {Partial<import('../../utils/types').ServerSettings>} */
+  let settings = {}
+  try {
+    settings = await detectServerSettings(devConfig, options, site.root)
+
+    cachedConfig.config = getConfigWithPlugins(cachedConfig.config, settings)
+  } catch (error_) {
+    log(NETLIFYDEVERR, error_.message)
+    exit(1)
+  }
+
+  command.setAnalyticsPayload({ projectType: settings.framework || 'custom', live: options.live, graph: options.graph })
+
+  log(`${NETLIFYDEVWARN} Building site for production`)
+
+  const { configPath: configPathOverride } = await runBuildTimeline({ cachedConfig, options, settings, site })
+
+  await startFunctionsServer({
+    api,
+    command,
+    config,
+    debug: options.debug,
+    loadDistFunctions: true,
+    settings,
+    site,
+    siteInfo,
+    siteUrl,
+    capabilities,
+    timeouts,
+  })
+
+  // Try to add `.netlify` to `.gitignore`.
+  try {
+    await ensureNetlifyIgnore(repositoryRoot)
+  } catch {
+    // no-op
+  }
+
+  // TODO: We should consolidate this with the existing config watcher.
+  const getUpdatedConfig = async () => {
+    const cwd = options.cwd || process.cwd()
+    const { config: newConfig } = await command.getConfig({ cwd, offline: true, state })
+    const normalizedNewConfig = normalizeConfig(newConfig)
+
+    return normalizedNewConfig
+  }
+
+  const inspectSettings = generateInspectSettings(options.edgeInspect, options.edgeInspectBrk)
+  const url = await startProxyServer({
+    addonsUrls,
+    config,
+    configPath: configPathOverride,
+    env,
+    geolocationMode: options.geo,
+    geoCountry: options.country,
+    getUpdatedConfig,
+    inspectSettings,
+    offline: options.offline,
+    settings,
+    site,
+    siteInfo,
+    state,
+  })
+
+  if (devConfig.autoLaunch !== false) {
+    await openBrowser({ url, silentBrowserNoneError: true })
+  }
+
+  process.env.URL = url
+  process.env.DEPLOY_URL = url
+
+  printBanner({ url })
+}
+
+/**
+ * Creates the `netlify serve` command
+ * @param {import('../base-command.mjs').default} program
+ * @returns
+ */
+export const createServeCommand = (program) =>
+  program
+    .command('serve')
+    .description('(Beta) build the site for production and serve locally')
+    .option(
+      '--context <context>',
+      'Specify a deploy context or branch for environment variables (contexts: "production", "deploy-preview", "branch-deploy", "dev")',
+      normalizeContext,
+    )
+    .option('-p ,--port <port>', 'port of netlify dev', (value) => Number.parseInt(value))
+    .option('--targetPort <port>', 'port of target app server', (value) => Number.parseInt(value))
+    .option('--framework <name>', 'framework to use. Defaults to #auto which automatically detects a framework')
```

Comment:
These aren't used

Reply:
Removed in https://github.com/netlify/cli/pull/5351/commits/42ae1bbd50b6a8e2866e1a1d65ad713e97b595aa.